### PR TITLE
Import limit validation

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Validate if the features limit will not be exceeded 
Screenshot 2023-09-12 at 14 08 17

### CODE REVIEW

File: src/lib/features/export-import-toggles/import-toggles-store.ts [hunk 1]
```diff
@@ -86,6 +90,38 @@ export class ImportTogglesStore implements IImportTogglesStore {
         return rows.map((row) => row.name);
     }
 
+    async getProjectFeaturesLimit(
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/import-toggles-store.ts:93]: this is the main algorithms

File: src/lib/features/export-import-toggles/import-toggles-store.ts [hunk 2]
```diff
@@ -86,6 +90,38 @@ export class ImportTogglesStore implements IImportTogglesStore {
...
+        const row = await this.db(T.projectSettings)
+            .select(['feature_limit'])
+            .where('project', project)
+            .first();
+        const limit: number = row?.feature_limit ?? Number.MAX_SAFE_INTEGER;
+
+        const existingFeaturesCount = await this.db(T.features)
+            .whereIn('name', featureNames)
+            .andWhere('project', project)
+            .where('archived_at', null)
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/import-toggles-store.ts:106]: exclude archived features

**Review by thomasheartman — APPROVED** [review]
Looks good to me 😄  But I remember that @sjaanus talked about limits and archived features at the offsite, so you should probably have him look over it too. That said, I suspect you've done the right thing here.

**Comment by sjaanus** [comment]
> Looks good to me 😄 But I remember that @sjaanus talked about limits and archived features at the offsite, so you should probably have him look over it too. That said, I suspect you've done the right thing here.

This is the result of this :smile:

File: src/lib/features/export-import-toggles/import-validation-messages.ts [hunk 3]
```diff
@@ -92,6 +95,16 @@ export class ImportValidationMessages {
                 affectedItems: [...featureNameCheckResult.invalidNames].sort(),
             });
         }
+        if (
+            featureLimitResult.currentFeaturesCount +
+                featureLimitResult.newFeaturesCount >
+            featureLimitResult.limit
+        ) {
+            errors.push({
+                message: `We detected you want to create ${featureLimitResult.newFeaturesCount} new features in a project with ${featureLimitResult.currentFeaturesCount} current features and a limit of ${featureLimitResult.limit}`,
```
Comment (sjaanus) [inline src/lib/features/export-import-toggles/import-validation-messages.ts:104]: Needed to read this 3 times, something is off. Better wording

We detected you want to create ${featureLimitResult.newFeaturesCount} new features to a project that already has ${featureLimitResult.currentFeaturesCount} existing features, exceeding the maximum limit of ${featureLimitResult.limit}.

**Review by sjaanus — APPROVED** [review]
One wording issue, other than that looks good

**Pull request merged**